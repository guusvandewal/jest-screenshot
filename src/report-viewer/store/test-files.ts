import { component } from "tsdi";
import { observable, action, computed } from "mobx";
import { FileReport } from "../../reporter-types";

@component
export class StoreTestFiles {
    @observable public activeFileName = testResults.files.length > 0 ? testResults.files[0].testFilePath : undefined;

    @action.bound public selectFile(file: FileReport) {
        this.activeFileName = file.testFilePath;
    }

    @computed public get activeFile() {
        return testResults.files.find(file => file.testFilePath === this.activeFileName);
    }

    public isActive(file: FileReport) {
        return this.activeFileName === file.testFilePath;
    }
}
