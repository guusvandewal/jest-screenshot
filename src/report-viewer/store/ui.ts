import { component } from "tsdi";
import { observable, action, computed } from "mobx";
import { FileReport } from "../../reporter-types";
import { bind } from "lodash-decorators";

@component
export class StoreUi {
    @observable public menuVisible = true;

    @bind @action public toggleMenu() {
        this.menuVisible = !this.menuVisible;
    }
}
