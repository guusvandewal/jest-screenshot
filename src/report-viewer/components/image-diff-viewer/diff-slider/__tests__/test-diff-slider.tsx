import * as React from "react";
import { shallow, mount } from "enzyme";
import { DiffSlider } from "..";
import { StoreUi } from "../../../../store";

describe("DiffSlider", () => {
    const someProps = {
        received: "reports/something/received.png",
        snapshot: "reports/something/snapshot.png",
        diff: "reports/something/diff.png",
        width: 100,
        height: 100,
    };

    beforeEach(() => {
        HTMLDivElement.prototype.getBoundingClientRect = () => ({
            top: 10, left: 10, right: 10, bottom: 10, width: 100, height: 100,
        } as DOMRect);
    });

    it("looks as expected", () => {
        expect(shallow(<DiffSlider {...someProps} />)).toMatchSnapshot();
    });

    [0, 1, 0.7].forEach(opacity => {
        it(`when setting the diff opacity to ${opacity}`, () => {
            const element = shallow(<DiffSlider {...someProps} />);
            const input = element.find("input[type='range']");
            input.simulate("change", { currentTarget: { value: opacity }});
            expect(element).toMatchSnapshot();
            expect(element.find("div.viewer-container div.viewer-diff img").prop("style").opacity).toBe(opacity);
        });
    });

    it("moves the slider when dragging on the image", () => {
        const element = mount(<DiffSlider {...someProps} />);
        const container = element.find("div.viewer-container");

        container.simulate("mousedown", { clientX: 60 });
        expect(element).toMatchSnapshot();
        expect(element.find("div.slider").prop("style").left).toBe("50%");
        expect(element.find("div.viewer-received").prop("style").width).toBe("50%");

        container.simulate("mousemove", { clientX: 85 });
        expect(element).toMatchSnapshot();
        expect(element.find("div.slider").prop("style").left).toBe("75%");
        expect(element.find("div.viewer-received").prop("style").width).toBe("75%");

        container.simulate("mouseup");
        container.simulate("mousemove", { clientX: 110 });
        expect(element).toMatchSnapshot();
        expect(element.find("div.slider").prop("style").left).toBe("75%");
        expect(element.find("div.viewer-received").prop("style").width).toBe("75%");
    });

    it("reacts to the window resizing", () => {
        const element = mount(<DiffSlider {...someProps} />);
        HTMLDivElement.prototype.getBoundingClientRect = () => ({
            top: 10, left: 10, right: 10, bottom: 10, width: 200, height: 100,
        } as DOMRect);
        window.dispatchEvent(new Event("resize"));
        expect((element.find("div.viewer-container").instance() as any).style.height).toBe("200px");
        [
            element.find("div.viewer-diff img"),
            element.find("div.viewer-received img"),
            element.find("div.viewer-snapshot img"),
        ].forEach(img => {
            const imgInstance: HTMLImageElement = img.instance() as any;
            expect(imgInstance.style.width).toBe("200px");
            expect(imgInstance.style.height).toBe("200px");
            expect(imgInstance.width).toBe(200);
            expect(imgInstance.height).toBe(200);
        });
    });

    it("reacts to the sidebar being toggled", async () => {
        const element = mount(<DiffSlider {...someProps} />);
        HTMLDivElement.prototype.getBoundingClientRect = () => ({
            top: 10, left: 10, right: 10, bottom: 10, width: 200, height: 100,
        } as DOMRect);
        tsdi.get(StoreUi).toggleMenu();
        await new Promise(resolve => setTimeout(resolve, 1));
        expect((element.find("div.viewer-container").instance() as any).style.height).toBe("200px");
        [
            element.find("div.viewer-diff img"),
            element.find("div.viewer-received img"),
            element.find("div.viewer-snapshot img"),
        ].forEach(img => {
            const imgInstance: HTMLImageElement = img.instance() as any;
            expect(imgInstance.style.width).toBe("200px");
            expect(imgInstance.style.height).toBe("200px");
            expect(imgInstance.width).toBe(200);
            expect(imgInstance.height).toBe(200);
        });
    });
});
