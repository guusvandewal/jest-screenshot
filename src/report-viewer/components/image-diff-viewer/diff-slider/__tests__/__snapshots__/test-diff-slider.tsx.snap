// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiffSlider looks as expected 1`] = `
<Fragment>
  <nav
    className="level settings"
  >
    <div
      className="level-item has-text-centered"
    >
      <div>
        <div
          className="heading"
        >
          Diff Opacity
        </div>
        <input
          className="title"
          max={1}
          min={0}
          onChange={[Function]}
          step="any"
          type="range"
          value={0.5}
        />
      </div>
    </div>
  </nav>
  <div
    className="viewer-container"
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
  >
    <div
      className="slider"
      style={
        Object {
          "left": "50%",
        }
      }
    >
      <div
        className="top"
      />
      <div
        className="bottom"
      />
    </div>
    <div
      className="viewer-diff"
    >
      <img
        src="reports/something/diff.png"
        style={
          Object {
            "opacity": 0.5,
          }
        }
      />
    </div>
    <div
      className="viewer-received"
      style={
        Object {
          "width": "50%",
        }
      }
    >
      <img
        src="reports/something/received.png"
      />
    </div>
    <div
      className="viewer-snapshot"
    >
      <img
        src="reports/something/snapshot.png"
      />
    </div>
  </div>
</Fragment>
`;

exports[`DiffSlider moves the slider when dragging on the image 1`] = `
<DiffSlider
  diff="reports/something/diff.png"
  height={100}
  received="reports/something/received.png"
  snapshot="reports/something/snapshot.png"
  width={100}
>
  <nav
    className="level settings"
  >
    <div
      className="level-item has-text-centered"
    >
      <div>
        <div
          className="heading"
        >
          Diff Opacity
        </div>
        <input
          className="title"
          max={1}
          min={0}
          onChange={[Function]}
          step="any"
          type="range"
          value={0.5}
        />
      </div>
    </div>
  </nav>
  <div
    className="viewer-container"
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
  >
    <div
      className="slider"
      style={
        Object {
          "left": "50%",
        }
      }
    >
      <div
        className="top"
      />
      <div
        className="bottom"
      />
    </div>
    <div
      className="viewer-diff"
    >
      <img
        src="reports/something/diff.png"
        style={
          Object {
            "opacity": 0.5,
          }
        }
      />
    </div>
    <div
      className="viewer-received"
      style={
        Object {
          "width": "50%",
        }
      }
    >
      <img
        src="reports/something/received.png"
      />
    </div>
    <div
      className="viewer-snapshot"
    >
      <img
        src="reports/something/snapshot.png"
      />
    </div>
  </div>
</DiffSlider>
`;

exports[`DiffSlider moves the slider when dragging on the image 2`] = `
<DiffSlider
  diff="reports/something/diff.png"
  height={100}
  received="reports/something/received.png"
  snapshot="reports/something/snapshot.png"
  width={100}
>
  <nav
    className="level settings"
  >
    <div
      className="level-item has-text-centered"
    >
      <div>
        <div
          className="heading"
        >
          Diff Opacity
        </div>
        <input
          className="title"
          max={1}
          min={0}
          onChange={[Function]}
          step="any"
          type="range"
          value={0.5}
        />
      </div>
    </div>
  </nav>
  <div
    className="viewer-container"
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
  >
    <div
      className="slider"
      style={
        Object {
          "left": "75%",
        }
      }
    >
      <div
        className="top"
      />
      <div
        className="bottom"
      />
    </div>
    <div
      className="viewer-diff"
    >
      <img
        src="reports/something/diff.png"
        style={
          Object {
            "opacity": 0.5,
          }
        }
      />
    </div>
    <div
      className="viewer-received"
      style={
        Object {
          "width": "75%",
        }
      }
    >
      <img
        src="reports/something/received.png"
      />
    </div>
    <div
      className="viewer-snapshot"
    >
      <img
        src="reports/something/snapshot.png"
      />
    </div>
  </div>
</DiffSlider>
`;

exports[`DiffSlider moves the slider when dragging on the image 3`] = `
<DiffSlider
  diff="reports/something/diff.png"
  height={100}
  received="reports/something/received.png"
  snapshot="reports/something/snapshot.png"
  width={100}
>
  <nav
    className="level settings"
  >
    <div
      className="level-item has-text-centered"
    >
      <div>
        <div
          className="heading"
        >
          Diff Opacity
        </div>
        <input
          className="title"
          max={1}
          min={0}
          onChange={[Function]}
          step="any"
          type="range"
          value={0.5}
        />
      </div>
    </div>
  </nav>
  <div
    className="viewer-container"
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
  >
    <div
      className="slider"
      style={
        Object {
          "left": "75%",
        }
      }
    >
      <div
        className="top"
      />
      <div
        className="bottom"
      />
    </div>
    <div
      className="viewer-diff"
    >
      <img
        src="reports/something/diff.png"
        style={
          Object {
            "opacity": 0.5,
          }
        }
      />
    </div>
    <div
      className="viewer-received"
      style={
        Object {
          "width": "75%",
        }
      }
    >
      <img
        src="reports/something/received.png"
      />
    </div>
    <div
      className="viewer-snapshot"
    >
      <img
        src="reports/something/snapshot.png"
      />
    </div>
  </div>
</DiffSlider>
`;

exports[`DiffSlider when setting the diff opacity to 0 1`] = `
<Fragment>
  <nav
    className="level settings"
  >
    <div
      className="level-item has-text-centered"
    >
      <div>
        <div
          className="heading"
        >
          Diff Opacity
        </div>
        <input
          className="title"
          max={1}
          min={0}
          onChange={[Function]}
          step="any"
          type="range"
          value={0}
        />
      </div>
    </div>
  </nav>
  <div
    className="viewer-container"
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
  >
    <div
      className="slider"
      style={
        Object {
          "left": "50%",
        }
      }
    >
      <div
        className="top"
      />
      <div
        className="bottom"
      />
    </div>
    <div
      className="viewer-diff"
    >
      <img
        src="reports/something/diff.png"
        style={
          Object {
            "opacity": 0,
          }
        }
      />
    </div>
    <div
      className="viewer-received"
      style={
        Object {
          "width": "50%",
        }
      }
    >
      <img
        src="reports/something/received.png"
      />
    </div>
    <div
      className="viewer-snapshot"
    >
      <img
        src="reports/something/snapshot.png"
      />
    </div>
  </div>
</Fragment>
`;

exports[`DiffSlider when setting the diff opacity to 0.7 1`] = `
<Fragment>
  <nav
    className="level settings"
  >
    <div
      className="level-item has-text-centered"
    >
      <div>
        <div
          className="heading"
        >
          Diff Opacity
        </div>
        <input
          className="title"
          max={1}
          min={0}
          onChange={[Function]}
          step="any"
          type="range"
          value={0.7}
        />
      </div>
    </div>
  </nav>
  <div
    className="viewer-container"
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
  >
    <div
      className="slider"
      style={
        Object {
          "left": "50%",
        }
      }
    >
      <div
        className="top"
      />
      <div
        className="bottom"
      />
    </div>
    <div
      className="viewer-diff"
    >
      <img
        src="reports/something/diff.png"
        style={
          Object {
            "opacity": 0.7,
          }
        }
      />
    </div>
    <div
      className="viewer-received"
      style={
        Object {
          "width": "50%",
        }
      }
    >
      <img
        src="reports/something/received.png"
      />
    </div>
    <div
      className="viewer-snapshot"
    >
      <img
        src="reports/something/snapshot.png"
      />
    </div>
  </div>
</Fragment>
`;

exports[`DiffSlider when setting the diff opacity to 1 1`] = `
<Fragment>
  <nav
    className="level settings"
  >
    <div
      className="level-item has-text-centered"
    >
      <div>
        <div
          className="heading"
        >
          Diff Opacity
        </div>
        <input
          className="title"
          max={1}
          min={0}
          onChange={[Function]}
          step="any"
          type="range"
          value={1}
        />
      </div>
    </div>
  </nav>
  <div
    className="viewer-container"
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
  >
    <div
      className="slider"
      style={
        Object {
          "left": "50%",
        }
      }
    >
      <div
        className="top"
      />
      <div
        className="bottom"
      />
    </div>
    <div
      className="viewer-diff"
    >
      <img
        src="reports/something/diff.png"
        style={
          Object {
            "opacity": 1,
          }
        }
      />
    </div>
    <div
      className="viewer-received"
      style={
        Object {
          "width": "50%",
        }
      }
    >
      <img
        src="reports/something/received.png"
      />
    </div>
    <div
      className="viewer-snapshot"
    >
      <img
        src="reports/something/snapshot.png"
      />
    </div>
  </div>
</Fragment>
`;
